external-crates/move/crates/move-compiler/src/expansion/translate.rs

```diff
@@ -1120,6 +1127,48 @@ fn check_visibility_modifiers(
     }
 }
 
+fn stdlib_definitions(context: &mut Context, mloc: Loc) -> StdlibDefinitions {
+    if maybe_make_well_known_address(context, mloc, stdlib_definitions::STDLIB_ADDRESS_NAME)
+        .is_none()
+    {
+        return StdlibDefinitions {
+            functions: BTreeMap::new(),
+            types: BTreeMap::new(),
+        };
+    }
+
+    // We don't want to report errors here if we can't find the thing.
+    let cur_diag_reporter = std::mem::replace(
+        &mut context.defn_context.reporter,
+        context.defn_context.env.dummy_diagnostic_reporter(),
+    );
```

Comment:
I am disturbed... but I guess this is okay? 
I do worry though that we will generate `String`s we don't need. But maybe it is fine here given how limited it is in scope and that this isn't really "hot" code (not in a loop and its once per module)

Reply:
The alternatives are: (1) write an an entire alternative execution path through the path expander that reports no errors and maintain the duplicated code, (2) rewrite all of path expansion to hand up a set of diagnostics to be reported, making that code far more complicated to support this single use case, or (3) passing a flag to the path expander to indicate if it should emit errors, meaning every path expansion error needs to do an extra flag check. This seemed preferable to (1), less-cumbersome than (2), and more-efficient than (3). I agree it is awful, though.

_(There is a fourth secret one where we add a "warning filter" for path expansion errors, but that way seemed to be further madness.)_